src/Sulu/Bundle/ContentBundle/Repository/ResourceLocatorRepository.php

```diff
@@ -71,7 +75,8 @@ public function generate($parts, $parentUuid, $uuid, $webspaceKey, $languageCode
 
         if ($parentUuid !== null) {
             $parentDocument = $this->documentManager->find($parentUuid, $languageCode, ['load_ghost_content' => false]);
-            $parentPath = $this->rlpStrategy->loadByContent($this->getPublishedAncestorOrSelf($parentDocument));
+            $publishedAncestorDocument = $this->documentInspector->getPublishedAncestorOrSelf($parentDocument);
+            $parentPath = $this->rlpStrategy->loadByContent($publishedAncestorDocument);
             $result = $this->rlpStrategy->generate($title, $parentPath, $webspaceKey, $languageCode, $segmentKey);
```

Comment:
You could pass here the `$parentUuid` instead of the `$parentPath`,  and do the `DocumentManager::find` in the `generate` method.